src/main/java/io/appium/java_client/touch/WebElementOption.java

```diff
@@ -0,0 +1,66 @@
+package io.appium.java_client.touch;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Optional.ofNullable;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.internal.HasIdentity;
+
+import java.util.Map;
+
+public class WebElementOption extends PositionOffsetOption<WebElementOption> {
+
+    private HasIdentity elementId;
+
+    /**
+     * This method creates a build instance of the {@link WebElementOption}
+     *
+     * @param element is the element to calculate offset from.
+     * @param x is the x-offset from the upper left corner of the given element.
+     * @param y is the y-offset from the upper left corner of the given element.
+     * @return the built option
+     */
+    public static WebElementOption elementOption(WebElement element, int x, int y) {
```

Comment:
I'd rather rename these to simply `element`, `relative` and `absolute`, so we can pass it like

`withOffset(element(el, x, y)), withOffset(element(el)), withOffset(relative(x,y)), withOffset(absolute(x,y))`

Reply:
@mykola-mokhnach 
I think it would be better to have two static methods
```java
element()
offset() //to not confuse user when he/she waits for an action with absolute offset but it acts with 
//relative or does the opposite. 
```

Reply:
If there was a flag that could make action use offset as some relative/absolute value forcefully I think that it would have sense.

Reply:
I think it's easier to perform the verification on the client side and throw an exception then if we explicitly define which type of offset we have - relative or absolute.